liquibase-service/src/main/resources/db/rdb/routines/007-sp_s_pagebuilder_postprocessing-002.sql

```diff
@@ -0,0 +1,1474 @@
+INSERT INTO [dbo].[job_flow_log]( batch_id, [Dataflow_Name], [package_Name], [Status_Type], [step_number], [step_name], [row_count] )
+VALUES( @Batch_id, @category, 'S_'+@category, 'START', @Proc_Step_no, @Proc_Step_Name, @RowCount_no );
+
+COMMIT TRANSACTION;
+
+BEGIN TRANSACTION;
+		SET @Proc_Step_no = 27;
+		SET @Proc_Step_Name = 'CREATE TABLE CODED_TABLE_MERGED_'+@category;
+		IF OBJECT_ID('#CODED_TABLE_MERGED_TEMP_INV', 'U') IS NOT NULL
+BEGIN
+DROP TABLE #CODED_TABLE_MERGED_TEMP_INV;
+END;
+		IF OBJECT_ID('#CODED_TABLE_MERGED_INV', 'U') IS NOT NULL
+BEGIN
+DROP TABLE #CODED_TABLE_MERGED_INV;
+END;
+
+SELECT temp_tbl.*
+INTO #CODED_TABLE_MERGED_INV
+FROM
+    (
+        SELECT [CODE_SET_GROUP_ID], [PAGE_CASE_UID], [NBS_QUESTION_UID], [NBS_CASE_ANSWER_UID], [ANSWER_TXT], [CODE_SET_NM], [RDB_COLUMN_NM], [ANSWER_OTH], [CODE], [ANSWER_TXT1], [answer_desc11], [ANSWER_TXT_CODE], [ANSWER_VALUE], [ANSWER_TXT2], [MASK], NULL AS OTHER_VALUE_IND_CD, NULL AS RDB_COLUMN_NM2
+        FROM #CODED_TABLE_DESC_INV
+        UNION ALL
+        SELECT [CODE_SET_GROUP_ID], [PAGE_CASE_UID], [NBS_QUESTION_UID], [NBS_CASE_ANSWER_UID], [ANSWER_TXT], [CODE_SET_NM], [RDB_COLUMN_NM], [ANSWER_OTH], [CODE], [ANSWER_TXT1], [answer_desc11], NULL, NULL, NULL, NULL, NULL, NULL
+        FROM #CODED_COUNTY_TABLE_DESC_INV
+        UNION ALL
+        SELECT [CODE_SET_GROUP_ID], [PAGE_CASE_UID], [NBS_CASE_ANSWER_UID], [NBS_QUESTION_UID], [ANSWER_TXT], NULL, [RDB_COLUMN_NM], NULL, NULL, NULL, [ANSWER_DESC11], NULL, NULL, NULL, NULL, [OTHER_VALUE_IND_CD], [RDB_COLUMN_NM2]
+        FROM #CODED_TABLE_OTHER_INV
+    ) AS temp_tbl;
+
+CREATE NONCLUSTERED INDEX [RDB_PERF_INTERNAL_04]
+		ON #CODED_TABLE_MERGED_INV
+		( [RDB_COLUMN_NM]
+		);
+
+SELECT @RowCount_no = @@ROWCOUNT;
+INSERT INTO [dbo].[job_flow_log]( batch_id, [Dataflow_Name], [package_Name], [Status_Type], [step_number], [step_name], [row_count] )
+VALUES( @Batch_id, @category, 'S_'+@category, 'START', @Proc_Step_no, @Proc_Step_Name, @RowCount_no );
+
+COMMIT TRANSACTION;
+
+if @debug = 'true'
+select * from #CODED_TABLE_MERGED_INV;
+
+
+BEGIN TRANSACTION;
+		SET @Proc_Step_no = 31;
+		SET @Proc_Step_Name = ' CREATE TABLE DATE_DATA_'+@category;
+		-- CREATE TABLE DATE_DATA AS
+		IF OBJECT_ID('#DATE_DATA_INV', 'U') IS NOT NULL
+BEGIN
+DROP TABLE #DATE_DATA_INV;
+END;
+
+SELECT NBS_CASE_ANSWER_UID, nrt_page.CODE_SET_GROUP_ID, nrt_page.RDB_COLUMN_NM,
+       (CASE
+            WHEN ISDATE(ANSWER_TXT) = 1 THEN ANSWER_TXT
+            ELSE NULL
+           END ) AS ANSWER_TXT1, ACT_UID AS PAGE_CASE_UID, nrt_page.RECORD_STATUS_CD,
+       nrt_page.NBS_QUESTION_UID, ( CASE
+                                        WHEN ISDATE(ANSWER_TXT) = 1 THEN ANSWER_TXT
+                                        ELSE NULL
+    END ) AS ANSWER_TXT,
+       nrt_page.INVESTIGATION_FORM_CD, CODE_SET_GROUP_ID AS CODE_SET_GROUP_ID1, QUESTION_GROUP_SEQ_NBR, DATA_TYPE
+INTO #DATE_DATA_INV
+FROM dbo.nrt_page_case_answer AS nrt_page WITH(NOLOCK) --4
```

Comment:
Since we have multiple nrt_page_case_answers in one stored procedure- I am wondering if it would be better to have one temporary table towards the start of the stored procedure with the max(last_chg_time) filtered in and that be replaced across the sproc. It might be more efficient than calling the subquery a couple of times.

Reply:
thank you and it makes sense. I have update updates to both the procedures.

Reply:
Additionally, including sp_hepatitis_datamart_postprocessing to this set.